tests/bdd/test_volume_create.py

```diff
@@ -0,0 +1,195 @@
+"""Volume creation feature tests."""
+import os
+import subprocess
+import time
+
+from pytest_bdd import (
+    given,
+    scenario,
+    then,
+    when,
+)
+
+import pytest
+import docker
+import requests
+
+import common
+
+from openapi.openapi_client.model.create_pool_body import CreatePoolBody
+from openapi.openapi_client.model.create_volume_body import CreateVolumeBody
+from openapi.openapi_client.model.volume_spec import VolumeSpec
+from openapi.openapi_client.model.protocol import Protocol
+from openapi.openapi_client.model.spec_state import SpecState
+from openapi.openapi_client.model.volume_state import VolumeState
+from openapi.openapi_client.model.volume_status import VolumeStatus
+
+VOLUME_UUID = "5cd5378e-3f05-47f1-a830-a0f5873a1449"
+VOLUME_SIZE = 10485761
+NUM_VOLUME_REPLICAS = 1
+REST_SERVER = "http://localhost:8081/v0"
+CREATE_REQUEST_KEY = "create_request"
+POOL_UUID = "4cc6ee64-7232-497d-a26f-38284a444980"
+NODE_NAME = "mayastor"
+
+
+# This fixture will be automatically used by all tests.
+# It starts the deployer which launches all the necessary containers.
+# A pool is created for convenience such that it is available for use by the tests.
+@pytest.fixture(autouse=True)
+def init():
+    common.deployer_start(1)
+    common.get_pools_api().put_node_pool(
+        NODE_NAME, POOL_UUID, CreatePoolBody(["malloc:///disk?size_mb=500"])
+    )
+    yield
+    common.deployer_stop()
+
+
+# Fixture used to pass the volume create request between test steps.
+@pytest.fixture(scope="function")
+def create_request():
+    return {}
+
+
+@scenario(
+    "features/volume/create.feature", "provisioning failure due to missing Mayastor"
+)
+def test_provisioning_failure():
+    """provisioning failure."""
+
+
+@scenario(
+    "features/volume/create.feature", "desired number of replicas cannot be created"
+)
+def test_desired_number_of_replicas_cannot_be_created():
+    """desired number of replicas cannot be created."""
+
+
+@scenario("features/volume/create.feature", "sufficient suitable pools")
+def test_sufficient_suitable_pools():
+    """sufficient suitable pools."""
+
+
+@given("a control plane, Mayastor instances and a pool")
+def a_control_plane_a_mayastor_instance_and_a_pool():
+    """a control plane, Mayastor instances and a pool."""
+    docker_client = docker.from_env()
+
+    # The control plane comprises the core agents, rest server and etcd instance.
+    for component in ["core", "rest", "etcd"]:
```

Comment:
could this be in a common fn? You could pass the container names as arguments

Reply:
Done.